etl/steps/data/meadow/artificial_intelligence/2024-06-03/epoch.py

```diff
@@ -0,0 +1,82 @@
+"""Load a snapshot and create a meadow dataset."""
+
+import numpy as np
+from structlog import get_logger
+
+from etl.helpers import PathFinder, create_dataset
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+def run(dest_dir: str) -> None:
+    log.info("epoch.start")
+
+    #
+    # Load inputs.
+    #
+    # Retrieve snapshot.
+    snap = paths.load_snapshot("epoch.csv")
+
+    # Read snapshot
+    tb = snap.read()
+
+    #
+    # Process data.
+    #
+    # Define columns of interest.
+    cols = [
+        "System",
+        "Domain",
+        "Authors",
+        "Country (from Organization)",
+        "Organization",
+        "Organization categorization",
+        "Publication date",
+        "Parameters",
+        "Training compute (FLOP)",
+        "Training dataset size (datapoints)",
+        "Notability criteria",
+    ]
+
+    # Check that the columns of interest are present
+    for col in cols:
+        assert col in tb.columns, f"Column '{col}' is missing from the dataframe."
+
+    # Select the columns of interest
+    tb = tb[cols]
+    # Replace empty strings with NaN values
+    tb.replace("", np.nan, inplace=True)
```

Comment:
In general, I would recommend either sticking always to `inplace` operations or non-inplace ones. My preference is not to use `inplace`. So:

```py
tb = tb.replace("", np.nan)